### PR TITLE
Improve Slack API behaviour on backend

### PR DESCRIPTION
**Pull request by mbalc**

### CODE REVIEW

File: server/src/main/java/com/omigost/server/notification/slack/SlackService.java [hunk 1]
```diff
@@ -27,13 +27,26 @@
     @Autowired
     private RestTemplate restTemplate;
 
+    private JsonNode getSlackBodyFrom(ResponseEntity<JsonNode> response) {
+        JsonNode body = response.getBody();
+
+        if (body == null) {
+            throw new RuntimeException("Null reponse from Slack API");
```
Comment (guser21) [inline server/src/main/java/com/omigost/server/notification/slack/SlackService.java]: Use specific exceptions such as Illegalargumentexception

File: server/src/main/java/com/omigost/server/notification/slack/SlackService.java [hunk 2]
```diff
@@ -27,13 +27,26 @@
...
+        }
+        if (!body.get("ok").asBoolean()) {
+            throw new RuntimeException("Slack API access failed, see these response details:\n" + body.toString());
```
Comment (guser21) [inline server/src/main/java/com/omigost/server/notification/slack/SlackService.java]: same here

Comment (mbalc) [reply server/src/main/java/com/omigost/server/notification/slack/SlackService.java]: I guess I can throw javax.ws.rs.WebApplicationException, but to descend lower in the hierarchy I'd have to have some class that constructs appropriate excetion based on status I guess, haven't found anything on that in 5 minutes

Comment (mbalc) [reply server/src/main/java/com/omigost/server/notification/slack/SlackService.java]: so considering that, what would you suggest here? 
refer to subclass tree available here
https://docs.oracle.com/javaee/7/api/javax/ws/rs/WebApplicationException.html

Comment (mbalc) [reply server/src/main/java/com/omigost/server/notification/slack/SlackService.java]: Update - those exceptions lower in hierarchy are not defined in `java.lang`

Comment (mbalc) [reply server/src/main/java/com/omigost/server/notification/slack/SlackService.java]: ok I see how can I implement that

File: server/src/main/java/com/omigost/server/notification/slack/SlackService.java (hunk 1 above)
Comment (mbalc) [reply server/src/main/java/com/omigost/server/notification/slack/SlackService.java]: done, plx resolve conversations

**Pull request merged**